api/src/main/java/io/opentelemetry/trace/SpanContext.java

```diff
@@ -75,25 +81,55 @@ public static SpanContext create(
    * @since 0.1.0
    */
   public static SpanContext createFromRemoteParent(
-      TraceId traceId, SpanId spanId, TraceFlags traceFlags, TraceState traceState) {
-    return new AutoValue_SpanContext(traceId, spanId, traceFlags, traceState, /* remote=*/ true);
+      String traceId, String spanId, TraceFlags traceFlags, TraceState traceState) {
+    return create(traceId, spanId, traceFlags, traceState, /* remote=*/ true);
   }
 
+  abstract String getTraceId();
+
+  abstract String getSpanId();
```

Comment:
return `CharSequence` instead?

Reply:
I think we discussed this, and decided to keep it as String for now. We can always open it up to CharSequence later, since it's a broader interface, I think.

Reply:
You can change return types that easily. If clients expect `String` they will break on `CharSequence`. You can widen input, not output

Reply:
true, but this isn't a public method!